File: mobile/lib/models/parish.dart
import 'enums.dart';

class Parish {
  final String id;
  final String dioceseId;
  final String parishCode;
  final String parishName;
  final String? patronSaint;
  final String? priestName;
  final String? establishedDate;
  final String? physicalAddress;
  final String? postalAddress;
  final String? contactEmail;
  final String? contactPhone;
  final String? bankAccountName;
  final String? bankAccountNumber;
  final String? bankName;
  final String? bankBranch;
  final String? mobileMoneyName;
  final String? mobileMoneyNumber;
  final String? mobileMoneyAccountName;
  final double? latitude;
  final double? longitude;
  final String? timezone;
  final bool? isActive;
  final String? createdAt;
  final String? updatedAt;

  Parish({
    required this.id,
    required this.dioceseId,
    required this.parishCode,
    required this.parishName,
    this.patronSaint,
    this.priestName,
    this.establishedDate,
    this.physicalAddress,
    this.postalAddress,
    this.contactEmail,
    this.contactPhone,
    this.bankAccountName,
    this.bankAccountNumber,
    this.bankName,
    this.bankBranch,
    this.mobileMoneyName,
    this.mobileMoneyNumber,
    this.mobileMoneyAccountName,
    this.latitude,
    this.longitude,
    this.timezone,
    this.isActive,
    this.createdAt,
    this.updatedAt,
  });

  factory Parish.fromJson(Map<String, dynamic> json) {
    return Parish(
      id: json['id'],
      dioceseId: json['diocese_id'],
      parishCode: json['parish_code'],
      parishName: json['parish_name'],
      patronSaint: json['patron_saint'],
      priestName: json['priest_name'],
      establishedDate: json['established_date'],
      physicalAddress: json['physical_address'],
      postalAddress: json['postal_address'],
      contactEmail: json['contact_email'],
      contactPhone: json['contact_phone'],
      bankAccountName: json['bank_account_name'],
      bankAccountNumber: json['bank_account_number'],
      bankName: json['bank_name'],
      bankBranch: json['bank_branch'],
      mobileMoneyName: json['mobile_money_name'],
      mobileMoneyNumber: json['mobile_money_number'],
      mobileMoneyAccountName: json['mobile_money_account_name'],
      latitude: json['latitude'] != null ? double.tryParse(json['latitude'].toString()) : null,
      longitude: json['longitude'] != null ? double.tryParse(json['longitude'].toString()) : null,
      timezone: json['timezone'],
      isActive: json['is_active'],
      createdAt: json['created_at'],
      updatedAt: json['updated_at'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'diocese_id': dioceseId,
      'parish_code': parishCode,
      'parish_name': parishName,
      'patron_saint': patronSaint,
      'priest_name': priestName,
      'established_date': establishedDate,
      'physical_address': physicalAddress,
      'postal_address': postalAddress,
      'contact_email': contactEmail,
      'contact_phone': contactPhone,
      'bank_account_name': bankAccountName,
      'bank_account_number': bankAccountNumber,
      'bank_name': bankName,
      'bank_branch': bankBranch,
      'mobile_money_name': mobileMoneyName,
      'mobile_money_number': mobileMoneyNumber,
      'mobile_money_account_name': mobileMoneyAccountName,
      'latitude': latitude,
      'longitude': longitude,
      'timezone': timezone,
      'is_active': isActive,
      'created_at': createdAt,
      'updated_at': updatedAt,
    };
  }
}

File: backend/src/models/parish.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use uuid::Uuid;
use chrono::{NaiveDate, DateTime, Utc};
use rust_decimal::Decimal;

#[derive(Debug, Serialize, Deserialize, FromRow)]
pub struct Parish {
    pub id: Uuid,
    pub diocese_id: Uuid,
    pub parish_code: String,
    pub parish_name: String,
    pub patron_saint: Option<String>,
    pub priest_name: Option<String>,
    pub priest_id: Option<Uuid>,
    pub established_date: Option<NaiveDate>,
    pub physical_address: Option<String>,
    pub postal_address: Option<String>,
    pub contact_email: Option<String>,
    pub contact_phone: Option<String>,
    pub bank_account_name: Option<String>,
    pub bank_account_number: Option<String>,
    pub bank_name: Option<String>,
    pub bank_branch: Option<String>,
    pub mobile_money_name: Option<String>,
    pub mobile_money_number: Option<String>,
    pub mobile_money_account_name: Option<String>,
    pub latitude: Option<Decimal>,
    pub longitude: Option<Decimal>,
    pub timezone: Option<String>,
    pub logo_url: Option<String>,
    pub is_active: Option<bool>,
    pub created_at: Option<DateTime<Utc>>,
    pub updated_at: Option<DateTime<Utc>>,
    pub deleted_at: Option<DateTime<Utc>>,
}

#[derive(Debug, Deserialize)]
pub struct CreateParishRequest {
    pub diocese_id: Uuid,
    pub parish_code: String,
    pub parish_name: String,
    pub patron_saint: Option<String>,
    pub priest_name: Option<String>,
    pub priest_id: Option<Uuid>,
    pub established_date: Option<NaiveDate>,
    pub physical_address: Option<String>,
    pub contact_email: Option<String>,
    pub contact_phone: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct UpdateParishRequest {
    pub parish_name: Option<String>,
    pub patron_saint: Option<String>,
    pub priest_name: Option<String>,
    pub priest_id: Option<Uuid>,
    pub physical_address: Option<String>,
    pub contact_email: Option<String>,
    pub contact_phone: Option<String>,
    pub is_active: Option<bool>,
}
